packages/edit-post/src/editor.native.js

```diff
@@ -144,15 +144,17 @@ class Editor extends Component {
 
 		return (
 			<SlotFillProvider>
-				<EditorProvider
-					settings={ editorSettings }
-					post={ normalizedPost }
-					initialEdits={ initialEdits }
-					useSubRegistry={ false }
-					{ ...props }
-				>
-					<Layout setTitleRef={ this.setTitleRef } />
-				</EditorProvider>
+				<UIStringsContext.Provider>
+					<EditorProvider
+						settings={ editorSettings }
+						post={ normalizedPost }
+						initialEdits={ initialEdits }
+						useSubRegistry={ false }
+						{ ...props }
+					>
+						<Layout setTitleRef={ this.setTitleRef } />
+					</EditorProvider>
+				</UIStringsContext.Provider>
```

Comment:
This will add the `uiStrings` prop all components inside the editor, right? If so, is there any issue in it being a prop on many components that don't need it?

I'm also thinking that for other devs not familiar with this project, it might not be clear that `uiStrings` is only for strings we wish to inject from outside Gutenberg, and that for the majority or user-facing strings it's not needed.

Reply:
You are completely right, and Context implementation here is not even needed. Just HOC.
Fixed on https://github.com/WordPress/gutenberg/pull/25238/commits/4be05a5e5e7d8ff1a8c46b9045cd12442bd6009d

Great catch!